pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -219,6 +219,14 @@ func (s *Reconciler) Update(ctx context.Context) error {
 				continue
 			}
 
+			// If the NIC has a failed provision operation, attempt to update it. Requeue on failure.
```

Comment:
Note: only the `Update` function needs to be updated, because it's called at the end of `Create`. This means handling the NIC provisioning failure here catches it for both creation and updating.